servicetalk-dns-discovery-netty/src/test/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererObserverTest.java

```diff
@@ -0,0 +1,266 @@
+        List<ResolutionResult> results = new ArrayList<>();
+        DnsClient client = dnsClient(__ -> name -> new NoopDnsResolutionObserver() {
+            @Override
+            public void resolutionCompleted(final ResolutionResult result) {
+                results.add(result);
+            }
+        });
+
+        Publisher<?> publisher = client.dnsQuery(HOST_NAME);
+        assertThat("Unexpected calls to resolutionComplete", results, hasSize(0));
+        // Wait until SD returns at least one address:
+        publisher.takeAtMost(1).ignoreElements().toFuture().get();
+        assertThat("Unexpected number of calls to resolutionComplete", results, hasSize(1));
+        ResolutionResult result = results.get(0);
+        assertThat(result.resolvedRecords(), is(2));
+        assertThat(result.ttl(), is(DEFAULT_TTL));
+        assertThat(result.becameActive(), is(2));
+        assertThat(result.becameInactive(), is(0));
+    }
+
+    @Test
+    public void srvQueryResolutionResult() throws Exception {
+        Map<String, ResolutionResult> results = new HashMap<>();
+        DnsClient client = dnsClient(__ -> name -> new NoopDnsResolutionObserver() {
+            @Override
+            public void resolutionCompleted(final ResolutionResult result) {
+                results.put(name, result);
+            }
+        });
+
+        Publisher<?> publisher = client.dnsSrvQuery(SERVICE_NAME);
+        assertThat("Unexpected calls to resolutionComplete", results.entrySet(), hasSize(0));
+        // Wait until SD returns at least one address:
+        publisher.takeAtMost(1).ignoreElements().toFuture().get();
+        assertThat("Unexpected number of calls to resolutionComplete", results.entrySet(), hasSize(2));
+
+        ResolutionResult srvResult = results.get(SERVICE_NAME);
+        assertThat(srvResult.resolvedRecords(), is(1));
+        assertThat(srvResult.ttl(), is(DEFAULT_TTL));
+        assertThat(srvResult.becameActive(), is(1));
+        assertThat(srvResult.becameInactive(), is(0));
+
+        ResolutionResult dnsResult = results.get(HOST_NAME + '.');
+        assertThat(dnsResult.resolvedRecords(), is(2));
+        assertThat(dnsResult.ttl(), is(DEFAULT_TTL));
+        assertThat(dnsResult.becameActive(), is(2));
+        assertThat(dnsResult.becameInactive(), is(0));
+    }
+
+    private static final class NoopDnsDiscoveryObserver implements DnsDiscoveryObserver {
+        static final DnsDiscoveryObserver INSTANCE = new NoopDnsDiscoveryObserver();
+
+        private NoopDnsDiscoveryObserver() {
+            // Singleton
+        }
+
+        @Override
+        public DnsResolutionObserver newResolution(final String name) {
+            return NoopDnsResolutionObserver.INSTANCE;
+        }
+    }
+
+    private static class NoopDnsResolutionObserver implements DnsResolutionObserver {
+        static final DnsResolutionObserver INSTANCE = new NoopDnsResolutionObserver();
+
```

Comment:
nit: add private constructor.

Reply:
This is not a final class, I extend it in a few tests.